src/input/types.js

```diff
@@ -112,21 +112,22 @@ export type BaseInputPropsT<T> = {
   onKeyUp?: (e: SyntheticKeyboardEvent<T>) => mixed,
   onFocus: (e: SyntheticFocusEvent<T>) => mixed,
   /** If true, adds a clear value icon button to the end of the input container. */
-  clearable: boolean,
+  clearable?: boolean,
+  onClear?: (e: SyntheticEvent<T>) => mixed,
   overrides: BaseInputComponentsT,
-  placeholder: string,
+  placeholder?: string,
   /** Renders component in 'required' state. */
   required: boolean,
   /** Renders component in provided size. */
   size: SizeT,
   /** Input type attribute. */
-  type: string,
+  type?: string,
   /** Input value attribute. */
   value?: string,
   rows?: number,
-};
+|};
 
-export type InputPropsT = {
+export type InputPropsT = {|
```

Comment:
this will break if additional props are provided to `Input`. While it will flag obvious errors that never affected the component it may be considered a breaking change since consumers will need to update their code.

Reply:
Well, we could consider it as a fixing a bug. If you are passing through props that have no effect on the Input component you might have a problem in your application?

Reply:
I would not be that strict (breaking change bump) when it comes to improving static types.

Reply:
hmm 105 also made object types exact by default right? because of that, i'm surprised this was required and it may be more impactful than a single component

Reply:
afaik, they just added a config option to enable them by default